### PR TITLE
support time current date function

### PR DESCRIPTION
**Pull request by siddontang**

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

File: expression/builtin/time.go [hunk 1]
```diff
@@ -301,6 +301,14 @@ func builtinSysDate(args []interface{}, ctx map[interface{}]interface{}) (interf
 	return builtinNow(args, ctx)
 }
 
+// See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_curdate
+func builtinCurrentDate(args []interface{}, ctx map[interface{}]interface{}) (interface{}, error) {
+	year, month, day := time.Now().Date()
+	return mysql.Time{
+		Time: time.Date(year, month, day, 0, 0, 0, 0, time.Local),
+		Type: mysql.TypeDate, Fsp: 0}, nil
```
Comment (qiuyesuifeng) [inline expression/builtin/time.go]: How about
```
mysql.Time{
		Time: time.Date(year, month, day, 0, 0, 0, 0, time.Local),
		Type: mysql.TypeDate, 
		Fsp: 0}, nil
```

Comment (siddontang) [inline expression/builtin/time.go]: seems no need.

File: parser/parser.y [hunk 2]
```diff
@@ -1940,6 +1942,16 @@ FunctionCallConflict:
 			return 1
 		}
 	}
+|	"CURRENT_DATE"
```
Comment (qiuyesuifeng) [inline parser/parser.y]: @shenli Does CURRENT_DATE function belong to FunctionCallConflict?

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**